server/service/src/sync/synchroniser.rs

```diff
@@ -26,6 +30,26 @@ pub struct Synchroniser {
     remote: RemoteDataSynchroniser,
 }
 
+#[derive(Error, Debug)]
+pub(crate) enum SyncError {
+    #[error("Database error while syncing")]
+    DatabaseError(#[from] RepositoryError),
```

Comment:
Added sync error vs anyhow return from synchroniser:
* Easier to map to sync api error (although can downcast but that's messy)
* Better use of thiserror annotation, previously in say RemotePushError, we would say "Sync api error in remote push" and we would add "remote push" to every variant. I think it's much cleaner to just add this once in the parent variant of underlying error